src/frontend/src/app/assets/assets.component.ts

```diff
@@ -7,14 +7,35 @@ import { AssetsService, AlertService } from '../services/index';
   styleUrls: ['./assets.component.css']
 })
 export class AssetsComponent implements OnInit {
+  public limit: Number = 0;
+  public offset: Number = 0;
   assets = [];
   assetsReadingsData = [];
+  asset_code;
   constructor(private assetService: AssetsService, private alertService: AlertService) { }
 
   ngOnInit() {
     this.getAsset();
   }
 
+  public setLimit(limit: Number) {
+    this.asset_code = <HTMLSelectElement>document.getElementById('code');
```

Comment:
move this line in getAssetReading method.

Reply:
Moved to `ngOnInit()` because need this variable in `setLimit` and `setOffset` methods to pass to `getAssetReading` method.